### DISCUSSION TITLE
Implement MPRIS protocol on Linux

### DISCUSSION THREAD

**Discussion by evgfilim1** [ppy/osu#17796] [Ideas] [answered]
osu! can be used as a music player when the user is in main menu (see awesome screenshot below).

![image](https://user-images.githubusercontent.com/17182159/163037243-0c42bbad-7236-4a2e-bc9a-a38903d5adc3.png)

However, you can't switch tracks without switching windows beforehand. So I think it would be nice to have MPRIS ([FreeDesktop specification](https://specifications.freedesktop.org/mpris-spec/latest/), [ArchWiki page](https://wiki.archlinux.org/title/MPRIS)) implemented to be able to switch tracks easily with some popular tools like [playerctl](https://github.com/altdesktop/playerctl). What do you think about it?

P.S. Unfortunately, quick googling of "c# mpris" doesn't yield something useful.

**Comment by bdach** [comment, marked as answer]
osu! is not a music player. This is out of scope.

**Reply by evgfilim1** [reply]
But it can be used like a player, so why not :thinking:

**Reply by bdach** [reply]
https://github.com/ppy/osu/search?q=%22osu%21+is+not+a+music+player%22&type=discussions

this has been a consistent stance for ages now. search around and you'll find your answer. the game being a music player for instance has implications (negative licencing implications) on the featured artists program.

**Reply by evgfilim1** [reply]
@bdach ok, got it, thanks.

For future readers, one more link: https://github.com/ppy/osu/discussions/16929#discussioncomment-2213814
